### PR TITLE
Network refactor

### PR DESCRIPTION
**Pull request by AxelVivienCosmo**

### CODE REVIEW

File: src/main/java/io/cosmosoftware/kite/instrumentation/Scenario.java [hunk 1]
```diff
@@ -7,148 +7,186 @@
...
-                      + "\r\n"
-                      + ReportUtils.getStackTrace(e));
-              result.append("  Error ").append(e.getMessage());
-            }
-            result.append(" on gateway ").append(gw).append("\n\n");
-          } else {
-            logger.info("No CleanUp to do on interface : " + inter + " on gateway " + gw);
-          }
-        }
+    String inter = "enp0s8";
```
Comment (ManuCosmo) [inline src/main/java/io/cosmosoftware/kite/instrumentation/Scenario.java]: why is this hardcoded here?

File: src/main/java/io/cosmosoftware/kite/instrumentation/Scenario.java [hunk 2]
```diff
@@ -7,148 +7,186 @@
...
+      SSHManager sshManager = new SSHManager(instance.getKeyFilePath(), instance.getUsername(),
+          instance.getIpAddress(), command);
+      if (sshManager.call().commandSuccessful()) {
+        Thread.sleep(1000);
+        result = "SUCCEEDED";
+      } else {
+        result = "FAILED";
+      }
+    } catch (Exception e) {
+      result = "thrown an ERROR";
```
Comment (ManuCosmo) [inline src/main/java/io/cosmosoftware/kite/instrumentation/Scenario.java]: what error?

File: src/main/java/io/cosmosoftware/kite/instrumentation/Scenario.java [hunk 3]
```diff
@@ -7,148 +7,186 @@
...
 
+  private String KiteServerCommand(String url) {
+    String result;
+    try {
+      URLConnection connection = new URL(url).openConnection();
+      connection.setRequestProperty("Accept-Charset", "UTF-8");
+      InputStream response = connection.getInputStream();
+      result = "SUCCEEDED and got response : " + response.toString();
+    } catch (Exception e) {
+      result = "thrown an ERROR";
```
Comment (ManuCosmo) [inline src/main/java/io/cosmosoftware/kite/instrumentation/Scenario.java]: result = "Error: " _+ e.get local message()

File: src/main/java/io/cosmosoftware/kite/instrumentation/NetworkProfile.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  /**
+   * Instantiates a new Instance.
+   *
+   * @param jsonObject the json object
+   */
+  public NetworkProfile(JsonObject jsonObject) throws KiteTestException {
+    String missingKey = "";
+    try {
+      missingKey = "name";
+      this.name = jsonObject.getString("name");
```
Comment (namvuCosmo) [inline src/main/java/io/cosmosoftware/kite/instrumentation/NetworkProfile.java]: if using missingKey then might as well use `this.name = jsonObject.getString(missingKey);`, or just call it key

File: src/main/java/io/cosmosoftware/kite/instrumentation/NetworkProfileHashMap.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+   *
+   * @param jsonArray the json array
+   */
+  public NetworkProfileHashMap(JsonArray jsonArray) throws KiteTestException {
+    for (int i = 0; i < jsonArray.size(); i++) {
+      try {
+        NetworkProfile network = new NetworkProfile(jsonArray.getJsonObject(i));
+        this.put(network.getName(), network);
+      } catch (KiteTestException e) {
+        throw e;
```
Comment (namvuCosmo) [inline src/main/java/io/cosmosoftware/kite/instrumentation/NetworkProfileHashMap.java]: so why catch it?

File: src/main/java/io/cosmosoftware/kite/instrumentation/Profile.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    this.duplicate = jsonObject.getInt("duplicate", 0);
+    this.bandwidth = jsonObject.getInt("bandwidth", 0);
+    this.command = jsonObject.containsKey("command") ? jsonObject.getString("command") : this.setCommand();
+  }
+
+  private String setCommand() throws Exception {
+    String command;
+    String egress_command = "";
+    String ingress_command = "";
+    command = "sudo ip link add ifb0 type ifb || true && sudo ip link set up dev ifb0 || true && sudo tc qdisc add dev " + this.nit + " ingress || true && sudo tc filter add dev " + this.nit + " parent ffff: protocol ip u32 match u32 0 0 action mirred egress redirect dev ifb0 || true && ";
```
Comment (namvuCosmo) [inline src/main/java/io/cosmosoftware/kite/instrumentation/Profile.java]: make this a bit more comprehensive for people who might work on your code after. Something like:
```
final String doThis = "this command do this";
final String doThat = "this command do that";
...

String command = doThis + " || " + doThat + " || " + ...
```

File: src/main/java/io/cosmosoftware/kite/instrumentation/Scenario.java (hunk 1 above)
Comment (AxelVivienCosmo) [reply src/main/java/io/cosmosoftware/kite/instrumentation/Scenario.java]: I don't really know how to handle interface for now ...

I think if I have an Instrumentation I take it from there, if the KiteServer create the grid I hardcode it assuming it is always the same and I have also manual command to handle

File: src/main/java/io/cosmosoftware/kite/instrumentation/NetworkProfile.java (hunk 4 above)
Comment (ManuCosmo) [reply src/main/java/io/cosmosoftware/kite/instrumentation/NetworkProfile.java]: the correct way should be to implement our own
getJsonString(jsonObject, "name");

```java
public static String getJsonString(JsonObject jsonObject, String key) throws KiteTestException {
  try {
    return jsonObject.getString("name");
  } catch (NullPointerException e) {
    throw new KiteTestException("Error in json config instrumentation, the key " + key+ " is missing.", Status.BROKEN, e);
  }
}
```

File: src/main/java/io/cosmosoftware/kite/instrumentation/Scenario.java (hunk 1 above)
Comment (ManuCosmo) [reply src/main/java/io/cosmosoftware/kite/instrumentation/Scenario.java]: ok, then declare at the top of the class. we'll probably change that at some point

```java
private static final String INTERFACE_0_NAME = "enp0s8";
private static final String INTERFACE_1_NAME = "enp0s9";
```

**Pull request merged**